### ISSUE TITLE
Improve user profile page

### ISSUE THREAD

**Issue by vinsdragonis** [vinsdragonis/Project-Nexus#73]
## What is needed?
The current user profile page seems to be pretty minimalistic. It would be nice if it could both show more details about the user and also display that user's posts on their profile page.

## What needs to be done?
- Modify the client-side page, and add required components to display as many essential details about a user as possible.
- Show that users posts on their profile page.

**Comment by Lydia02** [comment]
Hello, I would love to solve it
Did you mean this user profile UI?
![osp](https://user-images.githubusercontent.com/37791436/184559102-aab66378-b457-4882-b6b3-6144a1a4451b.png)

**Comment by vinsdragonis** [comment]
That panel certainly looks like it can get some enhancement. But what we need is more than just that. If you've noticed, you can currently only view your profile when you click on your profile pic.

We want to create a page that can similarly display other users' profiles and their posts. You could consider how Facebook designed its user profile pages. While we don't want to make a replica of that, you could consider that as a reference.

**Comment by vinsdragonis** [comment]
If you are interested, then you can go ahead with it @Lydia02

**Comment by Lydia02** [comment]
Great!
I would work on it.

**Comment by Lydia02** [comment]
The task is to create the a profile page, which will similarly show other users' post and profile.
## Task outline
1. Edit the profile.jsx and design the UI.
2. Add the home features i.e to be able to view other users' post and profile.
I am willing and ready to learn. Anyone willing to work me through this project? 
I am just learning reactJs.
Thank you.

**Comment by vinsdragonis** [comment]
Yes, you're right on that. But in order to fetch other user's data, you'll need to modify the routes on the server side. Then you can fetch it and display on the client application.

**Comment by Sam-Takahashi** [comment]
Hi. I'm trying to improve my reactJs skills. I mainly dabble in the frontend. I'd love to offer any help I can. @Lydia02

**Comment by Lydia02** [comment]
Okay, Great!
Hello @Sam-Takahashi, I don't mind.
Let's get it done together.

**Comment by vinsdragonis** [comment]
> Hi. I'm trying to improve my reactJs skills. I mainly dabble in the frontend. I'd love to offer any help I can. @Lydia02 

Sure, you can collaborate with @Lydia02 on this

**Comment by Lydia02** [comment]
Hello @Sam-Takahashi I have reached out already

**Comment by Lydia02** [comment]
> 



> Yes, you're right on that. But in order to fetch other user's data, you'll need to modify the routes on the server side. Then you can fetch it and display on the client application.

Hello @vinsdragonis ,in order to fetch other users' data, we would need the dotenv file or let me know what you think

**Comment by vinsdragonis** [comment]
> > 
> 
> > Yes, you're right on that. But in order to fetch other user's data, you'll need to modify the routes on the server side. Then you can fetch it and display on the client application.
> 
> Hello @vinsdragonis, in order to fetch other users' data, we would need the dotenv file or let me know what you think

Yes, you will need the dotenv file for this. I think you could create a separate copy of the DB on your local machine if possible. Else, I can share the MONGO_URL with you.

**Comment by Lydia02** [comment]
> 

You can kindly share the MONGO_URL with me
Thanks

**Comment by vinsdragonis** [comment]
mongodb+srv://guest:guest12321@cluster0.moo5s.mongodb.net/nexus?retryWrites=true&w=majority

**Comment by vinsdragonis** [comment]
@Lydia02 @Sam-Takahashi issue #72 involves part of what you're currently working on. I would request you to focus on that as well as it can solve both issues, and analyzing the work later during review becomes easier.

**Comment by Lydia02** [comment]
> @Lydia02 @Sam-Takahashi issue #72 involves part of what you're currently working on. I would request you to focus on that as well as it can solve both issues, and analyzing the work later during review becomes easier.

Okay Great!.
would do that also

**Comment by vikashjajra** [comment]
> > > 
> > 
> > 
> > > Yes, you're right on that. But in order to fetch other users' data, you'll need to modify the routes on the server side. Then you can fetch it and display it on the client application.
> > 
> > 
> > Hello @vinsdragonis, in order to fetch other users' data, we would need the dotenv file let me know what you think
> 
> Yes, you will need the dotenv file for this. I think you could create a separate copy of the DB on your local machine if possible. Else, I can share the MONGO_URL with you.

How do I create a copy of DB? or copy all the data to my own mongo_url so that whatever changes I make, won't break the original database, if that makes any sense?

**Comment by vinsdragonis** [comment]
> > > > 
> > > 
> > > 
> > > > Yes, you're right on that. But in order to fetch other users' data, you'll need to modify the routes on the server side. Then you can fetch it and display it on the client application.
> > > 
> > > 
> > > Hello @vinsdragonis, in order to fetch other users' data, we would need the dotenv file let me know what you think
> > 
> > Yes, you will need the dotenv file for this. I think you could create a separate copy of the DB on your local machine if possible. Else, I can share the MONGO_URL with you.
> 
> How do I create a copy of DB? or copy all the data to my own mongo_url so that whatever changes I make, won't break the original database, if that makes any sense?
> 

Hey! You can't replicate the whole database. And making your own database copy should be possible provided you've got through all the schemas we have until now. 
I, however, would highly recommend making the copy as a local version for experimental purposes only and switching back to the original one right after that, provided the code works fine. You can do this by changing the domain of the url to `https://localhost:3000`.
